toolkit/components/geometry/js/Geometry.jsm

```diff
@@ -96,25 +96,35 @@ class ArrayBufferDataStream {
       this.dataView.setUint32(this.pos, value);
       this.pos += 4;
     }
-  
-    writeUint8Array(value) {
-      for (const byte of value) {
-        this.writeUint8(byte);
+
+    writeString(value) {
+      const encoder = new TextEncoder();
+      // Note: in order to efficiently write this data, we first write the
+      // string data, reserving 4 bytes for the size.
+      const dest = new Uint8Array(this.dataView.buffer, this.pos + 4);
+      const encodeResult = encoder.encodeInto(value, dest);
+      if (encodeResult.read != value.length) {
```

Comment:
Will this be true if the string included unicode characters? And how does this work given that it gets encoded as utf8, when `String.length` gives the UTF16 code length? (genuinely curious haven't really thought much about it)

Reply:
The [docs describe this](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder/encodeInto#return_value) as `read` is the "number of UTF-16 units", which seems right to me.  They also mention comparing to length in the next sentence.